yarn start fails

```
[esbuild-scripts] Compiled successfully! [340ms]
[esbuild-scripts] Compiled successfully! [369ms]
[esbuild-scripts]
[esbuild-scripts] You can now view esbuild-app in the browser.
[esbuild-scripts]
[esbuild-scripts]   Local:            http://localhost:8000
[esbuild-scripts]   On Your Network:  http://192.168.31.128:8000
[esbuild-scripts]
[esbuild-scripts] Note that the development build is not optimized.
[esbuild-scripts] To create a production build, use yarn build.
[esbuild-scripts]
 > node_modules/@lukesheard/esbuild-scripts/build/runtime/index.js:7:22: error: Could not resolve "url" (use "platform: 'node'" when building for node)
    7 │ const url_1 = require("url");
      ╵                       ~~~~~

error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
```

--- a/.changeset/curly-rocks-notice.md
+++ b/.changeset/curly-rocks-notice.md
@@ -1,0 +1,5 @@
+---
+"@lukesheard/esbuild-scripts": patch
+---
+
+Fix issue where url is not resolved when using a built version of the packages.

--- a/packages/esbuild-scripts/src/plugins/incremental-compile.ts
+++ b/packages/esbuild-scripts/src/plugins/incremental-compile.ts
@@ -6,7 +6,6 @@ import * as fs from "fs/promises";
 import * as paths from "../config/paths";
 import * as logger from "../utils/logger";
 import { InstructionURLS } from "../config/urls";
-import { formatError } from "../utils/format-error";
 
 function createPlugin(urls: InstructionURLS): {
   plugin: Plugin;
@@ -52,6 +51,7 @@ function createPlugin(urls: InstructionURLS): {
             )}`
           );
         }
+
         if (isSuccessful && (process.stdout.isTTY || isFirstCompile)) {
           logger.log();
           logger.log(`You can now view ${chalk.bold(appName)} in the browser.`);
@@ -77,15 +77,6 @@ function createPlugin(urls: InstructionURLS): {
               `${chalk.cyan(`${useYarn ? "yarn" : "npm run"} build`)}.`
           );
           logger.log();
-        }
-
-        // If errors exist, only show errors.
-        if (result.errors.length) {
-          logger.log(chalk.red("Failed to compile.\n"));
-          // eslint-disable-next-line @typescript-eslint/no-misused-promises
-          result.errors.forEach(async (m) => {
-            logger.log(await formatError(m));
-          });
         }
 
         if (isFirstCompile) {

--- a/packages/esbuild-scripts/src/plugins/incremental-reporter.ts
+++ b/packages/esbuild-scripts/src/plugins/incremental-reporter.ts
@@ -1,0 +1,42 @@
+import { Plugin } from "esbuild";
+import chalk from "chalk";
+import * as logger from "../utils/logger";
+import { formatError } from "../utils/format-error";
+
+function createPlugin(): Plugin {
+  const plugin: Plugin = {
+    name: "incremental-errors",
+    setup(build) {
+      build.onEnd(async (result) => {
+        if (result.errors.length) {
+          logger.log(chalk.red("Failed to compile.\n"));
+        } else {
+          if (result.warnings.length) {
+            logger.log(chalk.red("Compiled with warnings.\n"));
+          }
+        }
+
+        // If errors exist, only show errors.
+        if (result.errors.length) {
+          await Promise.all(
+            result.errors.map(async (m) => {
+              logger.log(await formatError(m));
+            })
+          );
+        }
+
+        if (result.warnings.length) {
+          await Promise.all(
+            result.warnings.map(async (m) => {
+              logger.log(await formatError(m));
+            })
+          );
+        }
+      });
+    },
+  };
+
+  return plugin;
+}
+
+export default createPlugin;

--- a/packages/esbuild-scripts/src/runtime/index.ts
+++ b/packages/esbuild-scripts/src/runtime/index.ts
@@ -1,5 +1,4 @@
 import ErrorOverlay from "react-error-overlay";
-import { format } from "url";
 import stripAnsi from "strip-ansi";
 
 const isFirstCompilation: Record<string, boolean> = {};
@@ -34,16 +33,9 @@ ErrorOverlay.startReportingRuntimeErrors({
   filename: "/index.js",
 });
 
-const connection = new WebSocket(
-  format({
-    protocol: window.location.protocol === "https:" ? "wss" : "ws",
-    hostname: window.location.hostname,
-    port: window.location.port,
-    // Hardcoded in WebpackDevServer
-    pathname: "/_ws",
-    slashes: true,
-  })
-);
+const url = new URL("/_ws", window.location.href);
+url.protocol = url.protocol === "https:" ? "wss:" : "ws:";
+const connection = new WebSocket(url.toString());
 
 // Unlike WebpackDevServer client, we won't try to reconnect
 // to avoid spamming the console. Disconnect usually happens

--- a/packages/esbuild-scripts/src/start/index.ts
+++ b/packages/esbuild-scripts/src/start/index.ts
@@ -2,6 +2,7 @@ import * as esbuild from "esbuild";
 import chalk from "chalk";
 import * as express from "express";
 import ws from "express-ws";
+import * as fs from "fs/promises";
 import * as http from "http";
 import * as path from "path";
 import * as tmp from "tmp";
@@ -16,6 +17,7 @@ import getClientEnvironment, {
 import cssModulesPlugin from "../plugins/css-modules";
 import svgrPlugin from "../plugins/svgr";
 import incrementalCompilePlugin from "../plugins/incremental-compile";
+import incrementalReporterPlugin from "../plugins/incremental-reporter";
 import websocketReloadPlugin from "../plugins/ws-reload";
 
 import choosePort from "../utils/choose-port";
@@ -139,31 +141,56 @@ class DevServer {
   private hostRuntime = memoize(async () => {
     const runtimeDir = path.join(__dirname, "..", "runtime");
 
-    const result = await esbuild.build({
-      entryPoints: [runtimeDir],
-      bundle: true,
-      resolveExtensions: paths.moduleFileExtensions.map(
-        (extension) => `.${extension}`
-      ),
-      sourcemap: true,
-      absWorkingDir: paths.appPath,
-      format: "esm",
-      target: "es2015",
-      color: !isCi,
-      define: this.env.stringified,
-      watch: true,
-      write: true,
-      plugins: [websocketReloadPlugin("runtime", this.ws.getWss())],
-      outbase: runtimeDir,
-      outdir: path.join(this.outdir, "_runtime"),
-      publicPath: (await this.urls()).localUrlForBrowser,
-    });
+    const runtimeDirFiles = await fs.readdir(runtimeDir);
+    const indexFiles = runtimeDirFiles.filter((p) => p.startsWith("index"));
+    if (indexFiles.length !== 1) {
+      throw new Error(`Found multiple possible entry files`);
+    }
+    const entryPoint = indexFiles[0];
 
-    return result;
+    try {
+      return await esbuild.build({
+        entryPoints: [path.join(runtimeDir, entryPoint)],
+        bundle: true,
+        resolveExtensions: paths.moduleFileExtensions.map(
+          (extension) => `.${extension}`
+        ),
+        sourcemap: true,
+        absWorkingDir: paths.appPath,
+        format: "esm",
+        target: "es2015",
+        logLevel: "silent",
+        color: !isCi,
+        define: this.env.stringified,
+        watch: true,
+        write: true,
+        plugins: [
+          websocketReloadPlugin("runtime", this.ws.getWss()),
+          incrementalReporterPlugin(),
+        ],
+        outbase: runtimeDir,
+        outdir: path.join(this.outdir, "_runtime"),
+        publicPath: (await this.urls()).localUrlForBrowser,
+      });
+    } catch (e) {
+      const result = e as esbuild.BuildFailure;
+      logger.log(chalk.red("Failed to compile runtime.\n"));
+      const logs = result.errors.concat(result.warnings).map(async (m) => {
+        logger.log(await formatError(m));
+      });
+
+      await Promise.all(logs);
+
+      throw new Error(`Failed to compile runtime`);
+    }
   });
 
   private runEsbuild = async (watch: boolean) => {
-    const plugins: esbuild.Plugin[] = [cssModulesPlugin, svgrPlugin()];
+    const plugins: esbuild.Plugin[] = [
+      cssModulesPlugin,
+      svgrPlugin(),
+      incrementalReporterPlugin(),
+    ];
     let resolveIntialBuild;
     if (watch) {
       const { plugin, initialBuildPromise } = incrementalCompilePlugin(

